community/lucene-index/src/main/java/org/neo4j/kernel/api/impl/labelscan/LuceneLabelScanStore.java

```diff
@@ -114,9 +116,22 @@ else if ( !luceneIndex.isValid() )
         }
     }
 
+    private void create() throws IOException
```

Comment:
The whole idea before was to have index that are fully responsible for creation, flushing, closing etc to avoid index volatile variable and necessity of multiple index builders and state handling. With new way it's gone and we need to handle possible multiple indexes and always use volatile variable to read something from index. Maybe we can avoid doing that and come back to old way of doing things?